### PR TITLE
Added descriptions to the concept topics

### LINKED ISSUE #191: Concepts Page doesn't give any context on how garderner works
Documentation issue at https://gardener.cloud/docs/gardener/concepts/

Shouldn't the concepts page be more navigable. For someone new, I can't tell how to browse this.

### PR DESCRIPTION
**Pull request by n-boshnakov**
**How to categorize this PR?**

/area documentation
/kind enhancement

**What this PR does / why we need it**:
This PR adds short descriptions to the topics in the Concepts section.

**Which issue(s) this PR fixes**:
Fixes [#191](https://github.com/gardener/website-generator/issues/191) in gardener/website-generator

**Special notes for your reviewer**:

**Release note**:

```doc developer
NONE
```

### CODE REVIEW

File: docs/concepts/apiserver_admission_plugins.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ---
 title: APIServer Admission Plugins
+description: A list of all existing admission plugins together with their responsibilities
```
Comment (BeckerMax) [inline docs/concepts/apiserver_admission_plugins.md]:
```suggestion
description: A list of all gardener managed admission plugins together with their responsibilities
```

File: docs/concepts/architecture.md [hunk 2]
```diff
@@ -1,4 +1,6 @@
 ---
+title: Architecture
+description: The concepts behind the Kubernetes and Gardener architecture
```
Comment (BeckerMax) [inline docs/concepts/architecture.md]:
```suggestion
description: The concepts behind the Gardener architecture
```

File: docs/concepts/backup-restore.md [hunk 3]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Backup and Restore
+description: Understand the backup and restore configuration of Gardener
```
Comment (BeckerMax) [inline docs/concepts/backup-restore.md]:
```suggestion
description: Understand the etcd backup and restore capabilities of Gardener
```

File: docs/concepts/gardenlet.md [hunk 4]
```diff
@@ -1,5 +1,6 @@
 ---
 title: gardenlet
+description: Understand how Gardener can manage seed clusters decentrally
```
Comment (BeckerMax) [inline docs/concepts/gardenlet.md]:
```suggestion
description: Similarly to Kubernetes which runs a primary “agent” on every node, the kubelet, gardener runs a gardenlet as а primary “agent” on every seed cluster. Understand how the gardenlet works and learn more about the different Gardener components.
```

File: docs/concepts/resource-manager.md [hunk 5]
```diff
@@ -1,3 +1,8 @@
+---
+title: Resource Manager
+description: Learn more about the component that creates, updates, and deletes Kubernetes resources in a target cluster
```
Comment (BeckerMax) [inline docs/concepts/resource-manager.md]:
```suggestion
description: Learn more about the central component that exists once per gardener landscape and hosts controllers next to the gardener apiserver.
```

File: docs/concepts/resource-manager.md [hunk 6]
```diff
@@ -1,3 +1,8 @@
...
+---
```
Comment (BeckerMax) [inline docs/concepts/resource-manager.md:4]:
```suggestion
---
title: Gardener Resource Manager
description: Set of controllers with different responsibilities running once per Seed and once per Shoot.
---
```

File: docs/concepts/node-agent.md [hunk 7]
```diff
@@ -1,3 +1,8 @@
+---
+title: Node Agent
+description: Maintance of node-specific components without management by Kubernetes
```
Comment (BeckerMax) [inline docs/concepts/node-agent.md]:
```suggestion
description: How Gardener bootstraps machines into worker nodes and how it installs and maintains gardener-managed node-specific components.
```

**Review by BeckerMax — CHANGES_REQUESTED** [review]

**Comment by BeckerMax** [comment]
Maybe we should end all descriptions with a `point` at the end like K8s also does it https://kubernetes.io/docs/concepts/

**Comment by n-boshnakov** [comment]
> Maybe we should end all descriptions with a `point` at the end like K8s also does it https://kubernetes.io/docs/concepts/

The rest of the descriptions in our website do not end with periods, so adding them only to the Concepts section might feel off. I'd suggest we stick to not using them.

**Comment by n-boshnakov** [comment]
@BeckerMax I updated the `gardenlet` description to make it a bit shorter and took the second description for `Resource Manager`, while keeping the name the same in order to fit better with the rest of the topic titles. 

Regarding the titles, do you think that we should have them be generic or specific - `Resource Manager` vs `Gardener Resource Manager`?

**Comment by BeckerMax** [comment]
> Regarding the titles, do you think that we should have them be generic or specific - Resource Manager vs Gardener Resource Manager?

No strong preference. Internally we always include the `Gardener` as `Resource Manager` sounds very generic. Also short as in GRM. If you feel it is visually more appealing to use the shortened form that is fine for me too.

/lgtm

**Review by BeckerMax — APPROVED** [review]

**Comment by n-boshnakov** [comment]
> > Regarding the titles, do you think that we should have them be generic or specific - Resource Manager vs Gardener Resource Manager?
> 
> No strong preference. Internally we always include the `Gardener` as `Resource Manager` sounds very generic. Also short as in GRM. If you feel it is visually more appealing to use the shortened form that is fine for me too.
> 
> /lgtm

I can update the names for all the relevant topics to include "Gardener", that way it would look alright.

File: docs/concepts/admission-controller.md [hunk 8]
```diff
@@ -1,3 +1,8 @@
+---
+title: Admission Controller
+description: Functions and list of handlers for the Gardener Admission Controller
+---
+
 # Gardener Admission Controller
```
Comment (oliver-goetz) [inline docs/concepts/admission-controller.md]:
```suggestion
---
title: Gardener Admission Controller
description: Functions and list of handlers for the Gardener Admission Controller
---

## Overview
```
This one is not consistent with the others yet.

**Review by oliver-goetz — COMMENTED** [review]
Thanks for the PR 😄

**Pull request merged**